### PR TITLE
Add /auth handler to validate requests and comply with nginx auth_res…

### PR DESCRIPTION
**Pull request by EngHabu**
…ponse protocol

Signed-off-by: Haytham Abuelfutuh <haytham@afutuh.com>

# TL;DR
Implements `/auth` endpoint in flyte admin that can respond with 401, 403 or 202 depending on whether the request 

## Type
 - [ ] Bug Fix
 - [ ] Feature
 - [ ] Plugin

## Are all requirements met?

 - [ ] Code completed
 - [ ] Smoke tested
 - [ ] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
 _How did you fix the bug, make the feature etc. Link to any design docs etc_

## Tracking Issue
_Remove the '*fixes*' keyword if there will be multiple PRs to fix the linked issue_

fixes https://github.com/flyteorg/flyte/issues/1898

### CODE REVIEW

**Comment by EngHabu** [comment]
Closing since the behavior is identical to `/me` endpoint
